include/ginkgo/core/matrix/csr.hpp

```diff
@@ -100,26 +113,51 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
     friend class SparsityCsr<ValueType, IndexType>;
 
 public:
-    using EnableLinOp<Csr>::convert_to;
-    using EnableLinOp<Csr>::move_to;
-
     using value_type = ValueType;
     using index_type = IndexType;
     using mat_data = matrix_data<ValueType, IndexType>;
 
     class automatical;
 
+    /**
+     * strategy_type is to decide how to set the csr algorithm.
+     *
+     * The practical strategy method should inherit strategy_type and implement
+     * its `process`, `calc_size` function and the corresponding device kernel.
+     */
     class strategy_type {
         friend class automatical;
 
     public:
+        /**
+         * Creates a strategy_type.
+         *
+         * @param name  the name of strategy
+         */
         strategy_type(std::string name) : name_(name) {}
 
+        /**
+         * Returns the name of strategy
+         *
+         * @return the name of strategy
+         */
         std::string get_name() { return name_; }
 
+        /**
+         * Computes srow according to row pointers.
+         *
+         * @param mtx_row_ptrs  the row pointers of the matrix
+         * @param mtx_srow  the srow of the matrix
+         */
         virtual void process(const Array<index_type> &mtx_row_ptrs,
                              Array<index_type> *mtx_srow) = 0;
-
+        /**
+         * Computes the srow size according to the number of nonzeros.
+         *
+         * @param nnz  the number of nonzeros
+         *
+         * @return the size of srow
+         */
         virtual int64_t clac_size(const int64_t nnz) = 0;
```

Comment:
Was this function name supposed to be `clac_size` or `calc_size` ? Because in line 126 you say `calc_size` and I am not sure which is right ?